flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/hybrid/HsSubpartitionConsumerMemoryDataManager.java

```diff
@@ -173,7 +179,19 @@ public void releaseDataView() {
     @GuardedBy("consumerLock")
     private void trimHeadingReleasedBuffers() {
         while (!unConsumedBuffers.isEmpty() && unConsumedBuffers.peekFirst().isReleased()) {
-            unConsumedBuffers.removeFirst();
+            decreaseBacklog(unConsumedBuffers.removeFirst().getBuffer());
+        }
+    }
+
+    private void increaseBacklog(Buffer buffer) {
+        if (buffer.isBuffer()) {
+            backlog.getAndIncrement();
+        }
+    }
+
+    private void decreaseBacklog(Buffer buffer) {
```

Comment:
I believe that we should also handle this in `trimHeadingReleasedBuffers`.

Reply:
There exists `decreaseBacklog` in `trimHeadingReleasedBuffers ` currently. 🤔